src/communication/request_client_server.hpp

```diff
@@ -344,33 +346,48 @@ namespace cubcomm
   template <typename MsgId>
   css_error_code request_server<MsgId>::receive_request_buffer (size_t &message_size)
   {
-    size_t expected_size = 0;
-    size_t size_ilen = sizeof (expected_size);
-
     message_size = 0;
 
-    // NOTE: no ntohl here; integer value received as a stream of bytes
-    css_error_code err = m_channel.recv (reinterpret_cast <char *> (&expected_size), size_ilen);
-    if (err != NO_ERRORS)
+    size_t recv_size = m_recv_extensible_block.get_size ();
+    size_t rem_size = 0;
+    const css_error_code err = m_channel.recv_allow_truncated (m_recv_extensible_block.get_ptr (), recv_size,
+			       rem_size);
+    if (err == RECORD_TRUNCATED)
       {
-	er_log_recv_fail (m_channel, err);
-	return err;
-      }
-    assert (size_ilen == sizeof (expected_size));
+	// there is more data to retrieve
+	// extend the block and supply the new pointer
 
-    // will re-alloc until the block size gets big enough that will not neet to re-alloc anymore
-    m_recv_extensible_block.extend_to (expected_size);
+	assert (recv_size == m_recv_extensible_block.get_size ());
+	assert (rem_size > 0);
+
+	// TODO: maybe it should be extended to a multiple of page size?
+	m_recv_extensible_block.extend_by (rem_size);
```

Comment:
I suppose it would be better to extend to a multiple of the page size?

Reply:
I don't really know, but it seems that the communication buffer is irrelevant with the page size.